src/V3EmitCFunc.h

```diff
@@ -638,42 +638,66 @@ class EmitCFunc VL_NOT_FINAL : public EmitCConstInit {
         iterateChildrenConst(nodep);
     }
     void visit(AstCoverDecl* nodep) override {
-        putns(nodep, "vlSelf->__vlCoverInsert(");  // As Declared in emitCoverageDecl
-        puts("&(vlSymsp->__Vcoverage[");
-        puts(cvtToStr(nodep->dataDeclThisp()->binNum()));
-        puts("])");
-        // If this isn't the first instantiation of this module under this
-        // design, don't really count the bucket, and rely on verilator_cov to
-        // aggregate counts.  This is because Verilator combines all
-        // hierarchies itself, and if verilator_cov also did it, you'd end up
-        // with (number-of-instant) times too many counts in this bin.
-        puts(", first");  // Enable, passed from __Vconfigure parameter
-        puts(", ");
-        putsQuoted(protect(nodep->fileline()->filename()));
-        puts(", ");
-        puts(cvtToStr(nodep->fileline()->lineno()));
-        puts(", ");
-        puts(cvtToStr(nodep->offset() + nodep->fileline()->firstColumn()));
-        puts(", ");
-        putsQuoted((!nodep->hier().empty() ? "." : "")
-                   + protectWordsIf(nodep->hier(), nodep->protect()));
-        puts(", ");
-        putsQuoted(protectWordsIf(nodep->page(), nodep->protect()));
-        puts(", ");
-        putsQuoted(protectWordsIf(nodep->comment(), nodep->protect()));
-        puts(", ");
-        putsQuoted(nodep->linescov());
-        puts(");\n");
+        for (int i = 0; i < nodep->size(); i++) {
+            putns(nodep, "vlSelf->__vlCoverInsert(");  // As Declared in emitCoverageDecl
```

Comment:
Think we should make a new vlCoverToggleInsert that takes the width and index itself, and has the for loop itself.  Will help compile time further.